src/briefcase/integrations/java.py

```diff
@@ -74,38 +94,14 @@ def verify_install(cls, tools: ToolCache, install: bool = True, **kwargs) -> JDK
             return tools.java
 
         java = None
-        java_home = tools.os.environ.get("JAVA_HOME", "")
         install_message = None
 
-        # macOS has a helpful system utility to determine JAVA_HOME. Try it.
-        if not java_home and tools.host_os == "Darwin":
+        if java_home := tools.os.environ.get("JAVA_HOME", ""):
             try:
-                # If no JRE/JDK is installed, /usr/libexec/java_home
-                # raises an error.
-                java_home = tools.subprocess.check_output(
-                    ["/usr/libexec/java_home"],
-                ).strip("\n")
-            except subprocess.CalledProcessError:
-                # No java on this machine.
-                pass
-
-        if java_home:
-            try:
-                # If JAVA_HOME is defined, try to invoke javac.
-                # This verifies that we have a JDK, not a just a JRE.
-                output = tools.subprocess.check_output(
-                    [
-                        os.fsdecode(Path(java_home) / "bin" / "javac"),
-                        "-version",
-                    ],
-                )
-                # This should be a string of the form "javac 17.0.7\n"
-                version_str = output.strip("\n").split(" ")[1]
+                version_str = cls.jdk_version_from_path(tools, java_home)
                 if version_str == cls.JDK_RELEASE:
```

Comment:
This isn't testing the major version, but the exact version number. It looks like this was already incorrect, so we might as well fix it in this PR.

Extracting the major version isn't as simple as just taking the first number, because version 8 and earlier had a leading "1", e.g. "javac 1.8.0_242".

Reply:
Just verifying major version definitely makes sense to me. Right now, we're requiring users have the exact version of Java that Briefcase is using.

However, do we need to support parsing the versioning of Version 8? Since we're looking for `17`, this should just ignore any Version 8 installs of Java. Or am I missing something perhaps?

Reply:
I was imagining an error message saying "you have Java version 1". But it's better to keep the error message the way it is, with the full version number, because that's more helpful for debugging. In which case, you're right: the "1" prefix doesn't matter.